graphql_compiler/query_pagination/pagination_planning.py

```diff
@@ -33,8 +36,20 @@ def get_pagination_plan(schema_info, query_ast, number_of_pages):
     #                       - The root node has a unique index
     #                       - There are only a few different vertices at the root
     pagination_node = get_only_selection_from_ast(definition_ast, GraphQLError)
-    query_path = [pagination_node.name.value]
 
-    if pagination_node.name.value not in schema_info.pagination_keys:
+    pagination_field = schema_info.pagination_keys.get(pagination_node.name.value)
+    if pagination_field is None:
         return None
-    return PaginationPlan([VertexPartition(query_path, number_of_pages)])
+
+    if is_uniform_uuid4_type(schema_info, pagination_node.name.value, pagination_field):
+        pass
+    elif is_int_field_type(schema_info, pagination_node.name.value, pagination_field):
+        quantiles = schema_info.pagination_keys.statistics.get_field_quantiles(
+            pagination_node, pagination_field)
+        # We make sure there's more than enough quantiles because we don't interpolate.
+        if quantiles is None or len(quantiles) < 10 * number_of_pages:
```

Comment:
nit: move `len(quantiles) < 10 * number_of_pages` this is kind of hard on the eyes.

Reply:
where do you want it?

Reply:
Just move it to the line above.

Reply:
I can't check length if it's None